packages/flutter/lib/src/widgets/overflow_bar.dart

```diff
@@ -0,0 +1,482 @@
+    markNeedsLayout();
+  }
+
+  TextDirection get textDirection => _textDirection;
+  TextDirection _textDirection;
+  set textDirection(TextDirection value) {
+    if (_textDirection != value) {
+      _textDirection = value;
+      markNeedsLayout();
+    }
+  }
+
+  Clip get clipBehavior => _clipBehavior;
+  Clip _clipBehavior = Clip.none;
+  set clipBehavior(Clip value) {
+    assert(value != null);
+    if (value != _clipBehavior) {
+      _clipBehavior = value;
+      markNeedsPaint();
+      markNeedsSemanticsUpdate();
+    }
+  }
+
+  @override
+  void setupParentData(RenderBox child) {
+    if (child.parentData is! _OverflowBarParentData)
+      child.parentData = _OverflowBarParentData();
+  }
+
+  @override
+  double computeMinIntrinsicHeight(double width) {
+    RenderBox child = firstChild;
+    if (child == null)
+      return 0;
+    double height = 0.0;
+    while (child != null) {
+      height += child.getMinIntrinsicHeight(width);
+      child = childAfter(child);
+    }
+    return height + overflowSpacing * (childCount - 1);
+  }
+
+  @override
+  double computeMaxIntrinsicHeight(double width) {
+    RenderBox child = firstChild;
+    if (child == null)
+      return 0;
+    double height = 0.0;
+    while (child != null) {
+      height += child.getMaxIntrinsicHeight(width);
+      child = childAfter(child);
+    }
+    return height + overflowSpacing * (childCount - 1);
+  }
+
+  @override
+  double computeMinIntrinsicWidth(double height) {
+    RenderBox child = firstChild;
+    if (child == null)
+      return 0;
+    double width = 0.0;
+    while (child != null) {
+      width += child.getMinIntrinsicWidth(double.infinity);
+      child = childAfter(child);
+    }
+    return width + spacing * (childCount - 1);
+  }
+
+  @override
+  double computeMaxIntrinsicWidth(double height) {
+    RenderBox child = firstChild;
+    if (child == null)
+      return 0;
+    double width = 0.0;
+    while (child != null) {
+      width += child.getMaxIntrinsicWidth(double.infinity);
+      child = childAfter(child);
+    }
+    return width + spacing * (childCount - 1);
+  }
+
+  @override
+  double computeDistanceToActualBaseline(TextBaseline baseline) {
+    return defaultComputeDistanceToHighestActualBaseline(baseline);
+  }
+
+  @override
+  void performLayout() {
+    RenderBox child = firstChild;
+    if (child == null) {
+      size = constraints.smallest;
+      return;
+    }
+
+    final BoxConstraints childConstraints = constraints.loosen();
+    double childrenWidth = 0;
+    double maxChildHeight = 0;
+    double maxChildWidth = 0;
+
+    while (child != null) {
+      child.layout(childConstraints, parentUsesSize: true);
+      childrenWidth += child.size.width;
+      maxChildHeight = math.max(maxChildHeight, child.size.height);
+      maxChildWidth = math.max(maxChildWidth, child.size.width);
+      child = childAfter(child);
+    }
+
+    final bool rtl = textDirection == TextDirection.rtl;
```

Comment:
Nit:
```suggestion
    final bool isRtl = textDirection == TextDirection.rtl;
```

Reply:
I had it this way originally but I suffer from a personal loathing for logically capitalizing acronyms, so `rtl`.